fdpclient/client.py

```diff
@@ -260,6 +247,32 @@ def delete_distribution(self, id, **kwargs):
         self._request('delete', 'distribution', id=id, **kwargs)
 
     # Private methods
+    def _detect_fdp_url(self):
+        """Detect the internal path of fdp
+
+        Raises:
+            RuntimeError: failed to find the fdp url
+
+        Returns:
+            str: the internal path of fdp, i.e. 'fdp' or ''.
+        """
+        fmt = 'text/turtle'
+        while True:
```

Comment:
I think this while loop can be removed? You either break out on the first pass if one of the two options is detected, and otherwise you raise. So it'll never repeat?

Reply:
Yep, it'll never repeat.
It's used to reduce the number of requests. If the first request meets the conditions, then the second request will not be triggered, in this case we will save one-request time. So I'd like to keep the `while true`.

Reply:
Hi @LourensVeen, not sure if you see my last comment about keeping the `while True`. If you agree with that, I guess I can merge the PR then?

Reply:
Sorry, missed this, looks like a human error this time instead of a stray email filter, apologies.

I would much prefer an early return over a while loop that doesn't loop, like this:
```
fmt = 'text/turtle'

r = requests.get(self.host + '/fdp', params={'Accept':fmt})
if r.status_code == 200 and r.headers['content-type'] == fmt:
    return 'fdp'

r = requests.get(self.host, params={'Accept':fmt})
if r.status_code == 200 and r.headers['content-type'] == fmt:
    return ''

raise RuntimeError('Failed to detect the fdp url. Check if the '
        + 'server uses "<host>" or "<host>/fdp" as the fdp url.')
```

Reply:
I forgot where I got the wrong impression that there will be a linting warning when using multiple return. So I always tried to use one return at the end. But apparently using multiple returns in this case makes the code more clear. Thanks!